Discovery/Discovery/Presentation/DiscoveryView.swift

```diff
@@ -136,6 +138,13 @@ public struct DiscoveryView: View {
                     }
                     .frameLimit()
                 }.accessibilityAction {}
+                
+                if let model = Container.shared.resolve(StartupViewModel.self), !viewModel.userloggedIn {
+                    LogistrationBottomView(
+                        viewModel: model,
+                        sourceScreen: .discovery
+                    )
+                }
```

Comment:
We have to come up with something better to avoid modules circular dependency problems.
Feature modules do not need to know about each other.

Reply:
If LogistrationBottomView may be used across most application parts, maybe we should move it to the Core module? 🤔

Reply:
Maybe even without StartupViewModel 🤔
Let's make it simple/dumb View and handle it's button clicks outside with closures.
wdyt?

Reply:
> Maybe even without StartupViewModel 🤔
Let's make it simple/dumb View and handle it's button clicks outside with closures.
wdyt?

Yes, it's doable, but it will break the app style. Like we are not using closures anywhere in the app. 
We do have some places in the app where are already linking/importing feature modules e.g. `Discussion` is imported in `CourseUnitView` which is a part of `Course` module.

If you still want me to convert it to a dumb view please do let me know. I'll convert it.

Reply:
@volodymyr-chekyrta I've addressed other suggestions; only this point is left. Please let me know how you want to go with it.

Reply:
Thank you!

Discussion is different.
The Course module uses the Discussion module because of a conscious architectural decision, discussions are part of the course but were separated because of the large amount of functionality, we didn't want to bloat the Course module.
https://github.com/openedx/openedx-app-ios/blob/main/Documentation/Documentation.md#structure

> Like we are not using closures anywhere in the app.

That's not entirely true.
We have Views such as `StyledButton`, `SocialAuthButton`, `CustomDisclosureGroup`, and others. 
These Views all relay their actions via closures. 
In my perspective, `LogistrationBottomView` operates similarly; it's a simple view that should relay user actions back to the parent view.

Reply:
From my point of view, it will be correct to transform the `LogistrationBottomView` to a dumb view without ViewModel, move it to the Core module shared Views, and make it emit events to the parent.
But maybe I missed some context or part of the behavior in this View, and please correct me if I'm wrong.

@eyatsenkoperpetio @rnr Maybe you have ideas or suggestions? 
I would be happy to consider a few ideas before making a decision.

Reply:
@volodymyr-chekyrta I've made the LogistrationBottomView closure-based and moved it to Core. I've also added `authAnalytics.signUpClicked()` to the router in `showRegisterScreen(` because otherwise in Discovery we still have to import authorization module for analytics as `signUpClicked` is part of AuthorizationAnalytics.

Reply:
Thank you! Let me double check, but everything should be ok now.

Reply:
@saeedbashir Can we now remove the `Authorization.framework` from the `Discovery` module?
<img width="860" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/5f5280f7-b076-4e65-91c6-c90dc7aa9fb6">

Reply:
And also these imports for CourseDetailsView
<img width="624" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/8266bc34-b5e4-4a28-95a1-6934d9ff84ef">

Reply:
And I hope we can remove this one as well
<img width="572" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/20b7bbe1-f43e-4bf3-a14f-dc4e8b10d696">

Reply:
I don't know how it got there, but please also delete these two. They do nothing.
Thank you so much!
<img width="460" alt="image" src="https://github.com/openedx/openedx-app-ios/assets/127732735/d34ef936-90b8-4f58-ae24-5b2b8de2f40a">